trojsten/diplomas/sources.py

```diff
@@ -0,0 +1,195 @@
+                      'country': row.find('span', class_='results_country'),
+                      'team': row.find('span', class_='results_letter'),
+                      'participants': row.find('span', class_='results_participants')}
+            row_result = {}
+            for k, v in fields.items():
+                if v is None:
+                    continue
+                row_result[k] = v.text
+
+            if 'participants' in row_result:
+                participants = [x.strip() for x in row_result['participants'].split(',')]
+                for i, p in enumerate(participants):
+                    m = re.search(r'(.+) \([0-9]*\)', p.strip())
+                    if not m:
+                        break
+
+                    participants[i] = m.group(1)
+                row_result['participants'] = ", ".join(participants)
+
+            row_result['year'] = year
+            row_result['category'] = category
+            row_result['location'] = location
+            results.append(row_result)
+
+        return results
+
+    def handle_request(self, request, **kwargs):
+        request_body = request.body.decode('utf8')
+        kwargs.update(json.loads(request_body))
+        limit = kwargs.pop('limit', None)
+        limit = None if limit == '' else int(limit)
+        url = kwargs.pop('url', '')
+        url = url if url != '' else self.compose_url(**kwargs)
+        return self.scrape_results(url, limit)
+
+
+class NabojMath(Naboj):
+
+    name = 'naboj_math'
+
+    HOMEPAGE_URL = 'https://math.naboj.org'
+
+    def handle_request(self, request, **kwargs):
+        kwargs.update({
+            'pattern': "{homepage}/archive/results.php?".format(homepage=self.HOMEPAGE_URL) + '{args}',
+            'year': self.get_latest_year(self.HOMEPAGE_URL),
+            'country_code': 'sk'
+        })
+        return super(NabojMath, self).handle_request(request, **kwargs)
+
+    def render(self, **kwargs):
+        context = {
+            'categories': [{'name': 'Juniori', 'value': 'jun'}, {'name': 'Seniori', 'value': 'sen'}],
+            'name': self.name
+        }
+        context.update(kwargs)
+        template = loader.get_template('trojsten/diplomas/sources/naboj_math.html')
+        return template.render(context=context)
+
+
+class NabojPhysics(NabojMath):
+
+    name = 'naboj_physics'
+
+    HOMEPAGE_URL = 'https://physics.naboj.org'
+
+    def handle_request(self, request, **kwargs):
+        kwargs.update({
+            'pattern': "{homepage}/archive/results.php?".format(homepage=self.HOMEPAGE_URL) + '{args}',
+            'year': self.get_latest_year(self.HOMEPAGE_URL),
+            'country_code': 'sk'
+        })
+        return super(NabojPhysics, self).handle_request(request, **kwargs)
+
+    def render(self, **kwargs):
+        return super(NabojPhysics, self).render(name=self.name)
+
+
+SOURCE_CHOICES = [
+    (FileUpload.name, FileUpload.name),
```

Comment:
Predpokladam ze tu nejde pouzit class priamo, ci?

Ak nie tak toto vies jednoducho vygenerovat zo `SOURCE_CLASSES`.

Reply:
Toto tu priradujem ako choices do model fieldu, tam tie choices nie su tvorene on runtime, takze to asi lepsie nepojde.

Reply:
Myslim ze toto mohlo byt zapisane stylom:
```python
SOURCE_CHOICES = tuple((name, name) for name in SOURCE_CLASSES)
```
aby si to nemusel pisat 2x.